packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -24,7 +24,7 @@ const restrictedNodeFields = [
   `parent`,
 ]
 
-const restrictedContentTypes = [`entity`, `reference`]
+const restrictedContentTypes = [`entity`, `reference`, `tag`, `asset`]
```

Comment:
This is potentially breaking. Users having a `tag` contentType will no more be able to build their site. They might to be able to rewrite their queries and enable the `useNameForId ` option.


Asset is probably problematic since forever.


All restrictions can be lifted with the rewrites in #31286

Reply:
I wonder how many people use this?

Reply:
Asset probably nobody, as it am very certain it would break the build already.

Having a content type called `tag` might be used in very complex spaces. Still, it is much more likely its a field called `tag(s)`.

Reply:
Should we put this behind a flag then?